backend/ethereum/channel/funder.go

```diff
@@ -68,26 +77,44 @@ func NewFunder(backend ContractBackend) *Funder {
 	}
 }
 
-// copy returns a copy of the funder.
-func (f *Funder) copy() (_f *Funder) {
-	_f = NewFunder(f.ContractBackend)
-	for k, v := range f.accounts {
-		_f.accounts[k] = v
-	}
-	for k, v := range f.depositors {
-		_f.depositors[k] = v
+// RegisterAsset registers the depositor and account for the specified asset in
+// the funder.
+//
+// Deposits for this asset will be sent using the depositors from the
+// specified account when funding. Hence, it is the responsibility of the
+// caller to ensure, the account has sufficient balance in the asset.
+//
+// It returns true if the asset was successfully registered, false if it was already
+// present.
+func (f *Funder) RegisterAsset(asset Asset, d Depositor, acc accounts.Account) bool {
+	f.mtx.Lock()
+	defer f.mtx.Unlock()
+
+	// Both the maps (f.accounts & f.assets) are always modified togethe such
```

Comment:
It would probably be a good idea to merge them, right? Maybe something for another PR.

Reply:
Yes, I think it is better to merge these two fields. I think it could be done as a separate PR.